uk.co.compucorp.civicrm.tasksassignments/CRM/Tasksassignments/Reminder.php

```diff
@@ -486,6 +486,171 @@ private static function _getContactDailyReminderData($contactId, array $activity
     }
 
     /**
+     * Sends out daily email notifications for documents including separate lists
+     * with:
+     * - Overdue documents (!= approved)
+     * - Due in next fortnight (14 days) (!= approved)
+     * - Due in < 90 days (!= approved)
+     * and show status in email.
+     * 
+     * Acceptance criteria:
+     * - Emails are sent out to assignees only.
+     * - Assignee contact receives lists of documents he/she is assigned to only.
+     * - Documents are grouped by due date according to the logic described
+     *   in the task description (@see PCHR-1362).
+     * 
+     * Returns a number of emails sent.
+     * 
+     * @return int
+     */
+    public static function sendDocumentsNotifications() {
+      self::_setActivityOptions();
+      $count = 0;
+      $taSettings = civicrm_api3('TASettings', 'get');
+      $settings = $taSettings['values'];
+      $notifications = self::getDocumentNotificationsData();
+      $myTasksUrl = CIVICRM_UF_BASEURL . '/civicrm/tasksassignments/dashboard#/tasks/my';
+      $myDocumentsUrl = CIVICRM_UF_BASEURL . '/civicrm/tasksassignments/dashboard#/documents/my';
+      foreach ($notifications as $assignee => $documentsSet) {
+        list($assigneeId, $assigneeEmail) = explode(':', $assignee);
+        $templateBodyHTML = CRM_Core_Smarty::singleton()->fetchWith('CRM/Tasksassignments/Reminder/DailyDocumentsNotification.tpl', array(
+          'notification' => $documentsSet,
+          'baseUrl' => CIVICRM_UF_BASEURL,
+          'myTasksUrl' => $myTasksUrl,
+          'myDocumentsUrl' => $myDocumentsUrl,
+          'settings' => $settings,
+        ));
+        if (self::_send($assigneeId, $assigneeEmail, t('Documents Notification'), $templateBodyHTML)) {
+          $count++;
+        }
+      }
+
+      return $count;
+    }
+
+    /**
+     * Return an array containing a set of Documents for each Assignee
+     * grouped by due date period ('overdue' - overdue, 'plus14' - due date
+     * within next 14 days, 'plus90' - due date within next 90 days).
+     * 
+     * @return array
+     */
+    protected static function getDocumentNotificationsData() {
+      $today = new DateTime();
+      $todayPlus14 = (new DateTime())->add(new DateInterval('P14D'));
+      $todayPlus90 = (new DateTime())->add(new DateInterval('P90D'));
```

Comment:
"Documents due in next fortnight" - this is a description for 14 days variant in the template. Also, if I make the due dates as a parameter - it requires to assume that there are always two dates (for 14 and 90 days) - so it still will be not 100% flexible.